File: app/src/main/java/com/v1rex/smartincubator/Activities/BottonNavigationActivity.java
package com.v1rex.smartincubator.Activities;

import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.google.firebase.auth.FirebaseAuth;
import com.v1rex.smartincubator.Fragments.MentorsFragment;
import com.v1rex.smartincubator.Fragments.MessagesFragment;
import com.v1rex.smartincubator.Fragments.StartupsFragment;
import com.v1rex.smartincubator.R;

public class BottonNavigationActivity extends AppCompatActivity {
    private Toolbar toolbar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_botton_navigation);

        toolbar = (Toolbar) findViewById(R.id.my_toolbar);
        setSupportActionBar(toolbar);
        toolbar.setTitle(getString(R.string.app_name));
        toolbar.setSubtitle(getString(R.string.startups_name_fragment));

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
        bottomNavigationView.setOnNavigationItemSelectedListener(navListener);

        getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, new StartupsFragment()).commit();


    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.toolbar, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()){
            case R.id.action_settings:
//                startActivity(new Intent(BottonNavigationActivity.this, SettingsActivity.class));
                return true;
            case  R.id.action_search:
                startActivity(new Intent(BottonNavigationActivity.this, SearchActivity.class));
                return true;
            case R.id.action_logout:
                FirebaseAuth.getInstance().signOut();
                startActivity(new Intent(BottonNavigationActivity.this, MainActivity.class));

                default:
                    return super.onOptionsItemSelected(item);
        }



    }

    private BottomNavigationView.OnNavigationItemSelectedListener navListener =
            new BottomNavigationView.OnNavigationItemSelectedListener() {
                @Override
                public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    android.support.v4.app.Fragment selectedFragment = null;
                    switch(item.getItemId()){
                        case R.id.nav_startups:
                            selectedFragment = new StartupsFragment();
                            toolbar.setSubtitle(getString(R.string.startups_name_fragment));
                            break;
                        case R.id.nav_users:
                            selectedFragment = new MentorsFragment();
                            toolbar.setSubtitle(getString(R.string.mentors_name_fragments));
                            break;

                        case R.id.nav_messages:
                            selectedFragment = new MessagesFragment();
                            toolbar.setSubtitle(getString(R.string.messages_name_fragments));
                            break;
                    }

                    getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, selectedFragment).commit();

                    return true;

                }
            };
}

File: app/src/main/java/com/v1rex/smartincubator/Activities/BottonNavigationActivity.kt
package com.v1rex.smartincubator.Activities

import android.content.Intent
import android.support.design.widget.BottomNavigationView
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem

import com.google.firebase.auth.FirebaseAuth
import com.v1rex.smartincubator.Fragments.MentorsFragment
import com.v1rex.smartincubator.Fragments.MeetingsFragment
import com.v1rex.smartincubator.Fragments.MessagesFragment
import com.v1rex.smartincubator.Fragments.StartupsFragment
import com.v1rex.smartincubator.R
import kotlinx.android.synthetic.main.activity_botton_navigation.*

class BottonNavigationActivity : AppCompatActivity() {

    private val navListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        var selectedFragment: android.support.v4.app.Fragment? = null
        when (item.itemId) {
            R.id.nav_startups -> {
                selectedFragment = StartupsFragment()
                my_toolbar.subtitle = getString(R.string.startups_name_fragment)
            }
            R.id.nav_users -> {
                selectedFragment = MentorsFragment()
                my_toolbar.subtitle = getString(R.string.mentors_name_fragments)
            }

            R.id.nav_messages -> {
                selectedFragment = MessagesFragment()
                my_toolbar.subtitle = getString(R.string.messages_name_fragments)
            }

            R.id.nav_meetings -> {
                selectedFragment = MeetingsFragment()
                my_toolbar.subtitle = getString(R.string.meetings_name_fragments)
            }
        }

        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, selectedFragment).commit()

        true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_botton_navigation)

        // Setting the toolbar
        setSupportActionBar(my_toolbar as Toolbar)
        my_toolbar.title = getString(R.string.app_name)
        my_toolbar.subtitle = getString(R.string.startups_name_fragment)

        bottom_navigation.setOnNavigationItemSelectedListener(navListener)
        supportFragmentManager.beginTransaction().replace(R.id.fragment_container, StartupsFragment()).commit()

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.toolbar, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this@BottonNavigationActivity, SettingsActivity::class.java))
                return true
            }
            R.id.action_search -> {
                startActivity(Intent(this@BottonNavigationActivity, SearchActivity::class.java))
                return true
            }
            R.id.action_logout -> {
                FirebaseAuth.getInstance().signOut()
                startActivity(Intent(this@BottonNavigationActivity, LoginActivity::class.java))
                finish()
                return super.onOptionsItemSelected(item)
            }
            else -> return super.onOptionsItemSelected(item)
        }

    }

    override fun onBackPressed() {
        finish()
    }
}
